securedrop/journalist.py

```diff
@@ -242,9 +242,19 @@ def admin_reset_two_factor_hotp():
     otp_secret = request.form.get('otp_secret', None)
     if otp_secret:
         user = Journalist.query.get(uid)
-        user.set_hotp_secret(otp_secret)
-        db_session.commit()
-        return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        try:
+            user.set_hotp_secret(otp_secret)
+        except TypeError as e:
+            if "Non-hexadecimal digit found" in str(e):
+                flash("Invalid secret format: non-hexadecimal digit found")
```

Comment:
Note: many journalists will be unfamiliar with the term "hexadecimal digit". Perhaps something along the lines of: "Please only submit letters A-F and numbers 0-9"